docs/conf.py

```diff
@@ -8,12 +8,36 @@
 from sys import path
 
 from setuptools_scm import get_version
+from setuptools_scm.git import fetch_on_shallow
+from setuptools_scm.git import parse
 
 
+# -- Special accommodations for RTD ------------------------------------------
+
+
+def parse_with_fetch(*args, **kwargs) -> str:
+    """If the repo is found to be shallow, fetch a full.
+
+    By default, RTD does a fetch --limit 50, if a tag is not
+    present in the last 50 commits, the version reported by setuptools_scm
+    will be incorrect and appears as ``v0.1.dev...`` in the towncrier changelog.
+    Another approach is to enable ``DONT_SHALLOW_CLONE`` for the repo
+    https://docs.readthedocs.io/en/stable/feature-flags.html#feature-flags
+    This was done for ansible-navigator on the day of this commit.
```

Comment:
I think this part should probably be in a code comment. WDYT?

Reply:
I think the explanation is helpful in the docstring since it's such an odd thing
![image](https://user-images.githubusercontent.com/18386516/155760819-26e0508e-01d3-4a3a-8ca1-87dde7a468dc.png)

Reply:
Ah, okay. Usually, the motivation (the "why") goes into comments but the "how"/"what" goes into docstrings.